3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,59 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+
+
+async function fetchData(url) {
   // TODO complete this function
+  const response = await fetch(url);
+  const json = await response.json();
+  return json.results;
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
+
+function fetchAndPopulatePokemons(data, select) {
   // TODO complete this function
+  data.forEach((pokemon) => {
```

Comment:
Since we know that the 'data' is an array of 'pokemons' it is best to name it that way. Then this line could read:

```js
pokemons.forEach((pokemon) => {
```

It requires less cognitive 'mapping' to understand it. I don't have to think, it says 'data' but I know I must think 'pokemons'.